odxtools/cli/list.py

```diff
@@ -2,72 +2,93 @@
 import argparse
 from typing import Callable, List, Optional
 
+from rich import print
+
 from ..database import Database
 from ..diagcomm import DiagComm
 from ..diaglayer import DiagLayer
 from ..diagservice import DiagService
 from ..singleecujob import SingleEcuJob
 from . import _parser_utils
-from ._print_utils import format_desc, print_diagnostic_service
+from ._print_utils import format_desc, print_diagnostic_service, print_ecu_variant_metrics
 
 # name of the tool
 _odxtools_tool_name_ = "list"
 
 
-def print_summary(
-    odxdb: Database,
-    print_global_negative_responses: bool = False,
-    print_services: bool = False,
-    print_dops: bool = False,
-    print_params: bool = False,
-    print_comparams: bool = False,
-    print_pre_condition_states: bool = False,
-    print_state_transitions: bool = False,
-    print_audiences: bool = False,
-    allow_unknown_bit_lengths: bool = False,
-    variants: Optional[str] = None,
-    service_filter: Callable[[DiagComm], bool] = lambda x: True,
-) -> None:
-
-    diag_layer_names = variants if variants else [dl.short_name for dl in odxdb.diag_layers]
-
-    for dl_sn in diag_layer_names:
-        dl = odxdb.diag_layers[dl_sn]
-        if not dl:
-            print(f"The variant '{dl_sn}' could not be found!")
-            continue
+def print_summary(odxdb: Database,
+                  print_global_negative_responses: bool = False,
+                  print_services: bool = False,
+                  print_dops: bool = False,
+                  print_params: bool = False,
+                  print_comparam: bool = False,
+                  print_pre_condition_states: bool = False,
+                  print_state_transitions: bool = False,
+                  print_audiences: bool = False,
+                  allow_unknown_bit_lengths: bool = False,
+                  variants: Optional[List[str]] = None,
+                  service_filter: Callable[[DiagComm], bool] = lambda x: True,
+                  print_objects: bool = False) -> None:
+
+    diag_layer_names = [dl.short_name for dl in odxdb.diag_layers]
+    diag_layers: List[DiagLayer] = []
+
+    if variants is None:
+        variants = diag_layer_names
+
+    for name in variants:
+        if name in diag_layer_names:
+            diag_layers.append([x for x in odxdb.diag_layers if x.short_name == name][0])
+
+        else:
+            print(f"The variant '{name}' could not be found!")
+            return
+
+    if diag_layers:
+        print("\n")
+        print(f"Overview of diagnostic layers: ")
+        print_ecu_variant_metrics(diag_layers)
+
+    for dl in diag_layers:
+        print("\n")
+        print(f"[green]Diagnostic layer:[/green] '[bold white]{dl.short_name}[/bold white]'")
+        print(f" [blue]Variant Type[/blue]: {dl.variant_type.value}")
```

Comment:
```suggestion
        print(f"  [blue]Variant Type[/blue]: {dl.variant_type.value}")
```

Reply:
The old output `print(f"{dl.variant_type} '{dl.short_name}'")` was split into
`print(f"[green]Diagnostic layer:[/green] '[bold white]{dl.short_name}[/bold white]'")` and
 `print(f" [blue]Variant Type[/blue]: {dl.variant_type.value}")`
 Indentation was chosen to be the same as in `print(f" Description: " + desc)`